spring-cloud-gcp-stream/spring-cloud-gcp-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/PubSubMessageChannelBinder.java

```diff
@@ -0,0 +1,87 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.stream.binder.pubsub;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.gcp.pubsub.core.PubSubTemplate;
+import org.springframework.cloud.gcp.pubsub.support.SubscriberFactory;
+import org.springframework.cloud.gcp.stream.binder.pubsub.properties.PubSubConsumerProperties;
+import org.springframework.cloud.gcp.stream.binder.pubsub.properties.PubSubExtendedBindingProperties;
+import org.springframework.cloud.gcp.stream.binder.pubsub.properties.PubSubProducerProperties;
+import org.springframework.cloud.gcp.stream.binder.pubsub.provisioning.PubSubTopicProvisioner;
+import org.springframework.cloud.stream.binder.AbstractMessageChannelBinder;
+import org.springframework.cloud.stream.binder.ExtendedConsumerProperties;
+import org.springframework.cloud.stream.binder.ExtendedProducerProperties;
+import org.springframework.cloud.stream.binder.ExtendedPropertiesBinder;
+import org.springframework.cloud.stream.provisioning.ConsumerDestination;
+import org.springframework.cloud.stream.provisioning.ProducerDestination;
+import org.springframework.integration.core.MessageProducer;
+import org.springframework.integration.gcp.inbound.PubSubInboundChannelAdapter;
+import org.springframework.integration.gcp.outbound.PubSubMessageHandler;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHandler;
+
+/**
+ * @author João André Martins
+ */
+public class PubSubMessageChannelBinder
+		extends AbstractMessageChannelBinder<ExtendedConsumerProperties<PubSubConsumerProperties>,
+		ExtendedProducerProperties<PubSubProducerProperties>,
+		PubSubTopicProvisioner>
+	implements ExtendedPropertiesBinder<MessageChannel, PubSubConsumerProperties,
+		PubSubProducerProperties> {
+
+	@Autowired
+	private PubSubTemplate pubSubTemplate;
+
+	@Autowired
+	private SubscriberFactory subscriberFactory;
+
+	private PubSubExtendedBindingProperties pubSubExtendedBindingProperties =
+			new PubSubExtendedBindingProperties();
+
+	public PubSubMessageChannelBinder(boolean supportsHeadersNatively, String[] headersToEmbed,
+			PubSubTopicProvisioner provisioningProvider) {
+		super(supportsHeadersNatively, headersToEmbed, provisioningProvider);
+	}
+
+	@Override
+	protected MessageHandler createProducerMessageHandler(ProducerDestination destination,
+			ExtendedProducerProperties<PubSubProducerProperties> producerProperties)
+			throws Exception {
+		PubSubMessageHandler pubSubMessageHandler = new PubSubMessageHandler(this.pubSubTemplate);
+		pubSubMessageHandler.setTopic(destination.getName());
+
+		return pubSubMessageHandler;
+	}
+
+	@Override
+	protected MessageProducer createConsumerEndpoint(ConsumerDestination destination, String group,
```

Comment:
Same here. A subscription should be created, if it does no already exists. I think the topic could optionally also be created (subject to configuration)

Reply:
Did you mean `PubSubConsumerProperties`?

Reply:
Yes, this would be a property of PubSubConsumerProperties